transformer_engine/paddle/utils.py

```diff
@@ -73,12 +73,12 @@ def mask_to_cu_seqlens(mask: paddle.Tensor, need_kv: bool = False) -> paddle.Ten
     """Convert mask to cu_seqlens"""
     assert 'bool' in str(mask.dtype), "mask must be bool dtype"
     assert len(mask.shape) == 4 and mask.shape[1] == 1, "mask must be [b, 1, s_q, s_kv]"
-    q_actual_seqlens = paddle.sum(mask[:, :, :, 0] == False, axis=(-1, -2), dtype='int32')    # pylint: disable=singleton-comparison
+    q_actual_seqlens = paddle.sum(mask[:, :, :, 0] is False, axis=(-1, -2), dtype='int32')
```

Comment:
I think the `is` will evaluate to a single `False`, so the sum will always be 0.
```suggestion
    q_actual_seqlens = paddle.sum(mask[:, :, :, 0].logical_not(), axis=(-1, -2), dtype='int32')
```